### PR TITLE
Configurable basepath/url

### PR DESCRIPTION
**Pull request by rasmi-ranjan-guavus**
Description:

adds prefix to url in python and react

Steps to Check:

- configuring prefix in python
in __init__.py's PrefixMiddleware class update prefix
EX.  __init__(self, app, prefix = **'/demo'**):

- configuring prefix in webpack config
update URL_PREFIX in webpack.config.js
EX. URL_PREFIX = **'/demo'**;

### CODE REVIEW

File: superset/__init__.py [hunk 1]
```diff
@@ -39,7 +42,14 @@
 #################################################################
 # Handling manifest file logic at app start
 #################################################################
+
+def get_manifest_file() :
+  return APP_DIR + url_for('static', filename='assets/dist/manifest.json')
+
```
Comment (jitendra-kumawat) [inline superset/__init__.py]: how get_manifest_file used in project

File: superset/__init__.py [hunk 2]
```diff
@@ -160,11 +170,28 @@ def __call__(self, environ, start_response):
 for middleware in app.config.get('ADDITIONAL_MIDDLEWARE'):
     app.wsgi_app = middleware(app.wsgi_app)
 
+class PrefixMiddleware(object):
```
Comment (jitendra-kumawat) [inline superset/__init__.py]: class description ?

File: superset/__init__.py [hunk 3]
```diff
@@ -160,11 +170,28 @@ def __call__(self, environ, start_response):
...
+    def __call__(self, environ, start_response):
+
+       # if environ['PATH_INFO'].startswith(self.prefix):
+        environ['PATH_INFO'] = environ['PATH_INFO'][len(self.prefix):]
+
+        environ['SCRIPT_NAME'] = self.prefix
+        return self.app(environ, start_response)
+        # else:
+        #     start_response('404', [('Content-Type', 'text/plain')])
+        #     return ["This url does not belong to the app.".encode()]
```
Comment (jitendra-kumawat) [inline superset/__init__.py]: why if and else case are missed

File: superset/__init__.py [hunk 4]
```diff
@@ -160,11 +170,28 @@ def __call__(self, environ, start_response):
...
 
+app.wsgi_app = PrefixMiddleware(app.wsgi_app)
 class MyIndexView(IndexView):
     @expose('/')
     def index(self):
-        return redirect('/superset/welcome')
+        return redirect(url_for('Superset.welcome'))
```
Comment (jitendra-kumawat) [inline superset/__init__.py]: is it convention of url_for to pass value as  Superset.welcome instead /superset/welcome
 or we can use same here

File: superset/assets/spec/javascripts/sqllab/SqlEditorLeftBar_spec.jsx [hunk 5]
```diff
@@ -132,7 +132,7 @@ describe('SqlEditorLeftBar', () => {
         return d.promise();
       });
       wrapper.instance().fetchSchemas(1);
-      expect(ajaxStub.getCall(0).args[0]).to.equal('/superset/schemas/1/');
+      expect(ajaxStub.getCall(0).args[0]).to.equal('/rasmi/superset/schemas/1/');
       expect(wrapper.state().schemaOptions).to.have.length(3);
     });
     it('should handle error', () => {
```
Comment (jitendra-kumawat) [inline superset/assets/spec/javascripts/sqllab/SqlEditorLeftBar_spec.jsx]: we should use test/demo/guavus etc keyword instead particular name  in code

File: superset/assets/src/SqlLab/components/QuerySearch.jsx [hunk 6]
```diff
@@ -143,7 +143,7 @@ class QuerySearch extends React.PureComponent {
       this.state.to ? `to=${this.getTimeFromSelection(this.state.to)}` : '',
     ];
 
-    const url = this.insertParams('/superset/search_queries', params);
+    const url = this.insertParams('/rasmi/superset/search_queries', params);
     $.getJSON(url, (data, status) => {
```
Comment (jitendra-kumawat) [inline superset/assets/src/SqlLab/components/QuerySearch.jsx]: why rasmi path is hard coded

File: superset/assets/src/profile/components/RecentActivity.jsx [hunk 7]
```diff
@@ -27,7 +27,7 @@ export default class RecentActivity extends React.PureComponent {
           className="table table-condensed"
           mutator={mutator}
           sortable
-          dataEndpoint={`/superset/recent_activity/${this.props.user.userId}/?limit=${rowLimit}`}
+          dataEndpoint={`/rasmi/superset/recent_activity/${this.props.user.userId}/?limit=${rowLimit}`}
         />
       </div>
```
Comment (jitendra-kumawat) [inline superset/assets/src/profile/components/RecentActivity.jsx]: why rasmi path is hard coded . in all above file committed

File: superset/config.py [hunk 8]
```diff
@@ -47,7 +47,7 @@
 SUPERSET_CELERY_WORKERS = 32  # deprecated
 
 SUPERSET_WEBSERVER_ADDRESS = '0.0.0.0'
-SUPERSET_WEBSERVER_PORT = 8088
+SUPERSET_WEBSERVER_PORT = 8080
 SUPERSET_WEBSERVER_TIMEOUT = 60  # deprecated
```
Comment (jitendra-kumawat) [inline superset/config.py]: why port changed to 8080 it may conflict with  platform  ports

File: superset/config.py [hunk 9]
```diff
@@ -98,7 +98,13 @@
...
 
 # Uncomment to setup an App icon
-APP_ICON = '/static/assets/images/guavus_logo.svg'
+
+def get_app_icon():
+    return url_for('static', filename="assets/images/guavus_logo.svg")
+
+# APP_ICON = get_app_icon
+
+APP_ICON = "assets/images/guavus_logo.svg"
```
Comment (jitendra-kumawat) [inline superset/config.py]: can we use APP_ICON var in  get_app_icon method

File: superset/connectors/sqla/models.py [hunk 10]
```diff
@@ -281,7 +281,7 @@ class SqlaTable(Model, BaseDatasource):
     is_sqllab_view = Column(Boolean, default=False)
     template_params = Column(Text)
 
-    baselink = 'tablemodelview'
+    baseview = "TableModelView"
```
Comment (jitendra-kumawat) [inline superset/connectors/sqla/models.py]: why in above baselink changed to baseview and redfine its values

**Comment by arpit-agarwal** [comment]
This has been fixed by alternate implementation #207 & #208

**Pull request closed without merging**